From eab05f73b2d1f53915ac9b93f33f9a3d83e0cda2 Mon Sep 17 00:00:00 2001
From: Felix Yan <felixonmars@archlinux.org>
Date: Thu, 8 Aug 2024 22:06:25 +0300
Subject: [PATCH] test/cases/zsh-completion: Sync Linux banner fix from
 bash-completion

The same test failure appeared after
bd975862fa63bdb64efcd5c49c262e32a0d22380 updated the zsh-completion code. Let's apply the fix here too.
---
 test/cases/zsh-completion.bats | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/test/cases/zsh-completion.bats b/test/cases/zsh-completion.bats
index 500a756..9f02caa 100644
--- a/test/cases/zsh-completion.bats
+++ b/test/cases/zsh-completion.bats
@@ -14,7 +14,14 @@ setup() {
 
 @test "_detect_kver" {
     local kernel_ver tmp_knl
+    local arch
+    arch="$(uname -m)"
+
     kernel_ver="6.0.9-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 16 Nov 2022 17:01:17 +0000 x86_64 GNU/Linux"
+
+    # For non-i686/x86_64 platform, we rely on Linux banner for version detection.
+    # https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/init/version-timestamp.c?h=v6.7#n28
+    [[ "$arch" != @(i?86|x86_64) ]] && kernel_ver="Linux version ${kernel_ver}"
     tmp_knl=$(__gen_test_kernel "$kernel_ver")
     run _detect_kver "$tmp_knl"
     assert_output "6.0.9-arch1-1"
-- 
2.46.0

